specification/resources/resource-manager/Microsoft.Resources/stable/2024-03-01/deploymentStacks.json

```diff
@@ -0,0 +1,1456 @@
+      "post": {
+        "tags": [
+          "DeploymentStacks"
+        ],
+        "operationId": "DeploymentStacks_ExportTemplateAtManagementGroup",
+        "description": "Exports the template used to create the deployment stack.",
+        "x-ms-examples": {
+          "DeploymentStacksManagementGroupExportTemplate": {
+            "$ref": "./examples/DeploymentStackManagementGroupExportTemplate.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/ManagementGroupIdParameter"
+          },
+          {
+            "$ref": "#/parameters/DeploymentStackNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK - Returns the Template or TemplateLink payload of the deployment stack.",
+            "schema": {
+              "$ref": "#/definitions/DeploymentStackTemplateDefinition"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Resources/deploymentStacks/{deploymentStackName}/validate": {
+      "post": {
+        "tags": [
+          "DeploymentStacks"
+        ],
+        "operationId": "DeploymentStacks_PreflightValidateStackAtResourceGroup",
+        "description": "Runs preflight validation on the deployment stack template to verify its acceptance to Azure Resource Manager.",
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "azure-async-operation"
+        },
+        "x-ms-examples": {
+          "DeploymentStacksResourceGroupValidate": {
+            "$ref": "./examples/DeploymentStackResourceGroupValidate.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/DeploymentStackNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "deploymentStack",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/DeploymentStack"
```

Comment:
Why is DeploymentStack resource being defined here in request payload? The validate action is being performed on the same DeploymentStack resource here, right? SO, the service should already have access to resource, without needing it in the input payload.

Reply:
It is not. This endpoint is to do adhoc validation on a stack deployment without any side effects. A user can run this via the cli and the stacks service will do validation on the stack properties as well as send off the deployment portion to the `Microsoft.Resources/deployments`'s `validate` endpoint to do preflight on the deployment which involves running preflight on each deployment's resource. Preflight on a deployment's resource will call out to the RP of that resource to do basic validation on the resource inputs as defined in the deployment details. The stack itself in this context is stored as transient data. This allows to test iterations of a deployment, changes in deployment parameters, etc, quickly. Storing it transiently also supports testing a deployment stack with new inputs when the stack already exists as a resource. After validation is successful, a user can then PUT the stack again to apply the desired changes.